model/installation_request_test.go

```diff
@@ -49,12 +49,52 @@ func TestCreateInstallationRequestValid(t *testing.T) {
 				DNS:     "asupersupersupersupersupersupersupersupersupersupersupersuperlongname.com",
 			},
 		},
+		{
+			"DNS too short",
+			true,
+			&model.CreateInstallationRequest{
+				OwnerID: "owner1",
+				DNS:     "a.com",
+			},
+		},
+		{
+			"DNS starts with a hyphen",
+			true,
+			&model.CreateInstallationRequest{
+				OwnerID: "owner1",
+				DNS:     "-domain4321.com",
+			},
+		},
+		{
+			"DNS has invalid unicode characters",
+			true,
+			&model.CreateInstallationRequest{
+				OwnerID: "owner1",
+				DNS:     "letseat🍕.com",
+			},
+		},
+		{
+			"DNS has invalid special characters",
+			true,
+			&model.CreateInstallationRequest{
+				OwnerID: "owner1",
+				DNS:     "joram&gabe.com",
+			},
+		},
+		{
+			"DNS is a real FQDN that is in use already",
+			true,
+			&model.CreateInstallationRequest{
+				OwnerID: "owner1",
+				DNS:     "google.com",
+			},
```

Comment:
Off-topic comment: if this test ever fails then the internet has gotten very strange. ;)